File: js/shoppingCart.js
function ShoppingCart () {
  this.items = [];
}

ShoppingCart.prototype.addItem = function(_id, name, price, quantity) {
  if ( item = this._contains(_id) )
    item.quantity += 1;
  else
    this.items.push({_id: _id, name: name, price: price, quantity: quantity});
};

ShoppingCart.prototype.removeItem = function(_id) {
  for ( var i = 0; i < this.items.length; i++ )
    if (this.items[i]._id === _id) {
      if (this.items[i].quantity > 1) {
        this.items[i].quantity -=1;
        return true;
      }
      else {
        this.items.splice(i,1);
        return true;
      }
    }
  return false;
};

ShoppingCart.prototype.totalItems = function() {
  var total = 0;
  var i;
  for ( i = 0; i < this.items.length; i++)
    total += this.items[i].quantity
  return total;
};

ShoppingCart.prototype._contains = function(_id) {
  var i;
  for (i = 0; i < this.items.length; i++) {
    if ( this.items[i]._id === _id ) {
      return this.items[i];
    }
  }
  return false;
};

ShoppingCart.prototype.productQuantity = function(_id) {
  var i = 0;
  for (i = 0; i < this.items.length; i++) {
    if ( this.items[i]._id === _id ) {
      return this.items[i].quantity;
    }
  }
  return 0;
};

File: iOS/ClothesShop/ClothesShop/ShoppingCart.swift
//
//  ShoppingCart.swift
//  ClothesShop
//
//  Created by Sanda Golcea on 5/25/15.
//  Copyright (c) 2015 Sanda Golcea. All rights reserved.
//

import Foundation

class ShoppingCart {
    var items : [String : Item]
    
    init() {
        self.items = [String : Item]()
    }

    func addItem(product : Product) {
        if let item = self.items[product.id] {
            // checks that there still are products in stock before adding
            if (item.quantity < item.product.quantity) {
                item.quantity++
            }
        } else {
            var item = Item(product: product)
            self.items[product.id] = item
        }
    }
    
    func removeItem(product : Product) {
        if let item = self.items[product.id] {
            if item.quantity > 1 {
                item.quantity--
            } else {
                self.items[product.id] = nil
            }
        } else {
            self.items[product.id] = nil
        }
    }
    
    func totalItems() -> Int {
        var totalItems : Int = 0
        for (id, item) in self.items {
            totalItems += item.quantity
        }
        return totalItems
    }

    func totalPrice() -> Float {
        var total : Float = 0
        for (id, item) in self.items {
            total += Float(item.quantity) * item.product.price
        }
        return total
    }
    
    func getItems() -> [Item] {
        return self.items.values.array
    }
}
